lib/fluent/plugin/in_amqp.rb

```diff
@@ -121,10 +126,18 @@ def start
 
       q.subscribe(:manual_ack => @manual_ack) do |delivery, meta, msg|
         log.debug "Recieved message #{@msg}"
-        payload = parse_payload(msg)
+        body = parse_payload(msg)
+        if @manual_ack && @callback
+          cb = Proc.new do | success |
+            @ack_queue.push([delivery.delivery_tag, success])
+          end
+          payload = { :payload => payload, :callback => cb }
```

Comment:
confused where the callback is used

Reply:
over here: https://github.com/chronicled/fluent-plugin-output-graphql/pull/18

Since we are in the same process should be safe to pass a block (callback) around

obviously this is very implementation specific I would not put this code upstream